src/pages/challenges.tsx

```diff
@@ -29,20 +27,16 @@ const renderCardList = (
 
 const Challenges: FC = () => {
     const [filteredChallenge, setFilteredChallenge] = useState<any>([]);
-    const { data: user } = useGetUserQuery(skipToken);
-    const { response, error, isLoading, callAPI } = useFetch(
-        CHALLENGES_URL,
-        {},
-        false
-    );
+    const { data, isLoading, error } = useGetChallengesQuery();
+    
+    const challengesResponse = data?.challenges;
     const { state } = useAppContext();
     const { isLoading: isAuthenticating, isLoggedIn } = state;
 
     useEffect(() => {
-        if (isLoggedIn && !Object.keys(response).length) {
-            callAPI();
-            if ('challenges' in response) {
-                const challenges: challenge['content'] = response.challenges;
+        if (isLoggedIn && !Object.keys(challengesResponse).length) {
+            if ('challenges' in challengesResponse) {
+                const challenges: challenge['content'] = data.challenges;
                 const challengeMap: any = [];
                 challengeMap.Active = challenges.filter(
                     (task) => task.is_active
```

Comment:
this filtering of task can be done in transformResponse attribute provided by RTK query